### DISCUSSION TITLE
Proxmox

### DISCUSSION THREAD

**Discussion by iakat** [antonym/netboot.xyz#33] [Ideas] [answered]
It'd be nice to add Proxmox.
Link to latest: (4.0) http://www.proxmox.com/images/download/pve/iso/proxmox-ve_4.0-0d8559d0-17.iso

**Comment by Awmusic12635** [comment]
+1

**Comment by antonym** [comment]
It appears to be Debian based but unfortunately, this has the fun problem of not working well with memdisk ([possible fix](https://www.reversengineered.com/2016/01/07/booting-linux-isos-with-memdisk-and-ipxe/)).  I ran across this forum [post](https://forum.proxmox.com/threads/proxmox-installation-via-pxe-solution.8484/) but it involves modification of the original ISO which I'm trying to avoid doing.

**Comment by antonym** [comment]
Closing this unless someone wants to chase down the maintainer to get supported added to use with memdisk/iPXE.

**Comment by TomGlass** [comment]
Hello,

Wondering if a couple years later this is possible?

**Comment by salanki** [comment]
https://github.com/morph027/pve-iso-2-pxe would be nice to have that integrated

**Comment by antonym** [comment]
This would require the maintainer of the product to host the extracted ISO for netboot.xyz to leverage.  A lot of ISOs are possible to boot but iPXE isn't the best at mounting and loading files from the mount over the network.

**Comment by wrobelda** [comment]
@antonym pve-iso-2-pxe actually converts iso image to kernel and initrd images, so ISO booting is not required. Can you consider supporting it?

**Comment by thelamer** [comment]
Boot snippet here: 
```
#!ipxe
imgfree
set url https://github.com/thelamer/boot-assets/releases/download/proxmox/
kernel ${url}linux26 initrd=initrd
initrd ${url}initrd
boot
```
It is not hard to do the question is if we are greenlit to host these assets.

**Comment by orange888** [comment]
Please incorporate this. Thanks.

**Comment by Nickfost** [comment]
+1

**Comment by thelamer** [comment]
It is not a vote, it is a potential licensing issue. Just because is is possible does not make it legal IE windows.

**Comment by antonym** [comment]
Closing until we can determine if this is something we could host since it does have a subscription based licensing.

**Comment by LucaDev** [comment]
Proxmox is actually open source. It is released under the AGPL v3. Only support is subscription based.

**Comment by thelamer** [comment]
The problem is we are essentially rehosting their built assets from their website. The demo I showed above is not built from source. We would need to at a bare minimum get their permission to do this so we don't get a dmca.

**Comment by antonym** [comment]
I have sent Proxmox an e-mail to see if doing so would be considered acceptable.

**Comment by cjlittle** [comment]
I came across this discussion while looking for a solution to add Proxmox to my local netboot install.  I ended up hacking something together to get this to work, and have posted a brief HOW-TO and a simple proxmox.ipxe file in a [repo](https://github.com/cjlittle/netboot.xyz-proxmox).

**Comment by antonym** [comment]
Asset retriever built and this seemed to work well for an install without any modifications to the files... will work on getting the menu stitched together.

```
#!ipxe
imgfree
kernel https://github.com/netbootxyz/asset-mirror/releases/download/7.0-1-f4457cfd/vmlinuz vga=791 video=vesafb:ywrap,mtrr ramdisk_size=16777216 rw quiet initrd=initrd.magic
initrd https://github.com/netbootxyz/asset-mirror/releases/download/7.0-1-f4457cfd/initrd
initrd https://github.com/netbootxyz/asset-mirror/releases/download/7.0-1-f4457cfd/proxmox.iso /proxmox.iso
boot
```

Notes: To enable debug mode, replace quiet... splash=verbose proxdebug

**Comment by antonym** [comment, marked as answer]
https://github.com/netbootxyz/netboot.xyz/pull/950 Proxmox menu added to development branch.
